src/components/Header2.tsx

```diff
@@ -0,0 +1,104 @@
+import { useState } from 'react';
+import { GrClose } from 'react-icons/gr';
+import { GiHamburgerMenu } from 'react-icons/gi';
+import { Link } from 'react-router-dom';
+import { HiOutlineUserAdd } from 'react-icons/hi';
+import { IoIosLogIn, IoIosLogOut } from 'react-icons/io';
+import { CgProfile } from 'react-icons/cg';
+import { useAuth } from '../hooks';
+import { toast } from 'react-toastify';
+
+function Links() {
+  const { authenticated: isLoggedIn, setAuthenticated: setIsLoggedIn } =
+    useAuth();
+
+  const handleSignOut = () => {
+    setIsLoggedIn(false);
+  };
+
+  return (
+    <>
+      <Link to="/" className="hover:text-moonNeutral-600">
```

Comment:
Use useMemo to avoid unnecessary re-renders if the links don't change.

```
const authLinks = useMemo(() => (
    <>
      <Link to="./signup" title="Registration">
        <HiOutlineUserAdd className="text-2xl hover:text-moonNeutral-600" />
      </Link>
      <Link to="./signin" title="Login">
        <IoIosLogIn className="text-2xl hover:text-moonNeutral-600" />
      </Link>
    </>
  ), []);
```